### PR TITLE
Fix UIOpacity issue on native platform

### PR DESCRIPTION
**Pull request by yoki0805**
Issue:
[UIOpacity doesn't effect the child node when it becomes active the first time in native.](https://github.com/cocos/cocos-engine/issues/17419)

Testing in creator 3.8.4 community
1. Web (Correct)
web
2. Simulator (Wrong)
sim
3. After the fix
sim-fixed

Test demo:
[NewProject.zip](https://github.com/user-attachments/files/16509730/NewProject.zip)

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

## Greptile Summary

This PR addresses an issue where UIOpacity doesn't affect child nodes when they become active for the first time on native platforms, improving opacity handling and propagation.

- Modified `cocos/2d/components/ui-opacity.ts` to check if a node has been loaded before accessing UI properties
- Updated local opacity setting for render entities to ensure proper application
- Adjusted the recursive opacity setting process to handle newly activated child nodes correctly
- Improved compatibility between web and native platforms for UIOpacity behavior
- Fixed potential performance issues related to unnecessary opacity updates

### CODE REVIEW

File: cocos/2d/components/ui-opacity.ts [hunk 1]
```diff
@@ -125,14 +127,22 @@ export class UIOpacity extends Component {
...
         if (uiOp && stopRecursiveIfHasOpacity) {
             // Because it's possible that UiOpacity components are handled by themselves (at onEnable or onDisable)
             uiOp._parentOpacity = parentOpacity;
             return;
         }
+
+        // If the node has never been loaded, then node._uiProps.uiComp won't be set.
+        // It has to be obtained by getComponent at this time.
+        let render = null as UIRenderer | null;
+        if (node._objFlags & IsOnLoadCalled) {
```
Comment (qiuguohua) [inline cocos/2d/components/ui-opacity.ts:140]: Is it possible to use the _isOnLoadCalled interface directly?

Comment (yoki0805) [reply cocos/2d/components/ui-opacity.ts:140]: Probably not. The `_isOnLoadCalled` interface belongs to `Component`.
I can't guarantee `uiOp` or `render` will be there at this point.

Comment (qiuguohua) [reply cocos/2d/components/ui-opacity.ts:140]: ok，got it.

**Review by qiuguohua — APPROVED** [review]

**Review by minggo — APPROVED** [review]

File: cocos/2d/components/ui-opacity.ts (hunk 1 above)
Comment (yoki0805) [reply cocos/2d/components/ui-opacity.ts:140]: I'm sorry about that.
It's some mistake here. The `node._objFlags & IsOnLoadCalled` always be false.
I will create another PR to fix it.
@qiuguohua

Comment (yoki0805) [reply cocos/2d/components/ui-opacity.ts:140]: [Fix the UIRenderer check logic in UIOpacity.](https://github.com/cocos/cocos-engine/pull/17508)
@qiuguohua
@minggo

**Pull request merged**